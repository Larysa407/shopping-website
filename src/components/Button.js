import React from 'react'
import "./Button.scss"

export default function Button({backgroundColor, text, onClick, disabled}) {
  return (
    <button className="btn" style={{ backgroundColor }} disabled={disabled} onClick={onClick}>
    {text}
  </button>

  )
}
