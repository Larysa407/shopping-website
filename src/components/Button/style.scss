.btn{
    width: 80px;
    height: 30px;
    border-radius: 20px;
    border: 1px solid lightslategrey;
    box-shadow: 0 3px 4px darkslategray;
    font-size: 20px;
    font-size: 12px;
    cursor: pointer;
    
    &:hover{
        background-color: rgb(155, 165, 165);
    }
}

.confirmBtn,
.cancelBtn{
    text-decoration: none;
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;
}
