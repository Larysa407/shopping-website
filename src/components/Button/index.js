import React from 'react'
import "./style.scss"

export default function Button({backgroundColor, text, onClick, disabled}) {
  return (
    <button className="btn" style={{ backgroundColor }} disabled={disabled} onClick={onClick}>
    {text}
  </button>

  )
}
