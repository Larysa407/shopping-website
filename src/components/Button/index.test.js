import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Button from "./index";
import "@testing-library/jest-dom/extend-expect";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../../redux/store/index"

describe("Button component", () => {
  test("test for text and background color of button", () => {
    const buttonText = "To cart";
    const backgroundColor = "yellow";
    const onClick = jest.fn();

    const { getByText } = render(
      <Button
        backgroundColor={backgroundColor}
        text={buttonText}
        onClick={onClick}
      />
    );

    const buttonElement = getByText(buttonText);

    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveStyle({ background: backgroundColor });
  });

  test("calls onClick function when button is clicked", () => {
    const onClick = jest.fn();

    const { getByText } = render(
      <Button backgroundColor="blue" text="To cart" onClick={onClick} />
    );

    const buttonElement = getByText("To cart");

    fireEvent.click(buttonElement);

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  test("snapshotTest for Button component", () => {
      const component = renderer.create(
        <Provider store={store}>
          <Router>
            <Button />
          </Router>
        </Provider>
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
});
