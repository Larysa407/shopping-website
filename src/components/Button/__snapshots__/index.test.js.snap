// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshotTest for Button component 1`] = `
<button
  className="btn"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  Button
</button>
`;
